ui/src/app/components/app-router.tsx

```diff
@@ -56,37 +60,24 @@ export const NavRedirect = ({path}) => {
   return null;
 };
 
-export const AppRoute = ({path, data = {}, guards = [], component: Component, exact = true}): React.ReactElement => {
-  const routeParams = useParams();
-  const routeHistory = useHistory();
-
-  return <Route exact={exact} path={path} render={
-    () => {
-      const { redirectPath = null } = fp.find(({allowed}) => !allowed(), guards) || {};
-      return redirectPath
-        ? <NavRedirect path={redirectPath}/>
-        : <Component urlParams={routeParams} routeHistory={routeHistory} routeConfig={data}/>;
-    }}>
-  </Route>;
-};
+interface AppRouteProps extends RouteProps {
+  guards?: Array<Guard>;
+}
 
-export const ProtectedRoutes = (
-  {guards, children}: {guards: Guard[], children: React.ReactElement | React.ReactElement[] }): React.ReactElement => {
+export class AppRoute extends Route<AppRouteProps> {
```

Comment:
I am a bit concerned with this approach because it creates a dependency on an implementation detail of `Route`, namely that it is a class component. If a future version of react-router switched this to be a function component, we break and don't have a path to recreating this approach.

What were the issues with using `AppRoute` directly in the switch? Or just using `Route` at the top-level only?

Reply:
Using `AppRoute` directly in the switch only worked because of duck-typing; react-router looks for something with a `path` prop. 

Using `Route` at the top level would work, it would just be significantly more verbose. Each guarded route would have to replicate L75:L78 within its render prop.

Reply:
Wait, I think I see what you're saying. It probably still works if I only use `Route` in `app-router` and then don't use another `<Switch>` in `signed-in-app-routing` and `workspace-app-routing`, because the `Switch` only cares about its immediate children and `SignedIn` or `WorkspaceWrapper` is ultimately rendering the lower routes.

Reply:
I actually think it would even work without `Route`ifying all the top-level things because there's only one layer of `ProtectedRoutes` and the `ProtectedRoutes` is last.

Reply:
I mean only at the top-level - none of said routes are guarded.

Another option to consider is to switch to a more config-driven approach. This would allow us to insert a `Route` at the top level, and nest a guarded component beneath, which may address both issues of boilerplate and Switch functionality. Example template: https://reactrouter.com/web/example/route-config ; thoughts?

Reply:
Sorry -d idn't see your updated comments before I submitted that ^

Yes - that is what I meant

Reply:
Cool. I'm going to do the minimal change of adding the `Switch` in `app-router`.